transport/tchannel/inbound.go

```diff
@@ -0,0 +1,73 @@
+package tchannel
+
+import (
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/internal/introspection"
+	"go.uber.org/yarpc/internal/sync"
+)
+
+// Inbound receives YARPC requests over TChannel. It may be constructed using
+// the NewInbound method on a tchannel.Transport.
+type Inbound struct {
+	once      sync.LifecycleOnce
+	transport *Transport
+}
+
+// NewInbound returns a new TChannel inbound backed by a shared TChannel
+// transport.
+// There should only be one inbound for TChannel since all outbounds send the
+// listening port over non-ephemeral connections so a service can deduplicate
```

Comment:
we had a conversation a while back about allowing multiple tchannel inbounds but having all but one be ephemeral, do we still want to do that?